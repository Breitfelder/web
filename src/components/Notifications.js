import React from 'react';
import styled from 'styled-components';

const NotificationList = styled.div`
  position: absolute;
  bottom: 0;
  right: 0;
  max-width: 50%;
  z-index: 9999;
`;

const BaseNotification = styled.div`
  position: relative;
  border-radius: 5px;
  padding: 1rem;
  margin: 0 1rem 1rem 1rem;
`;

const NotificationLine = styled.p`
  padding: 0;
  margin: 0;

  & + & {
    margin-top: 0.5rem;
  }
`;

export var CloseButton = styled.button`
  font-size: 1.5rem;
  position: absolute;
  right: 0.5rem;
  top: 0.2rem;
  padding: 0.3rem 0.5rem;
  margin: 0;
  border: none;
  background: transparent;
  cursor: pointer;
`;

const ErrorNotification = BaseNotification.extend`
  background: #f44336;
  color: white;
  box-shadow: 0 4px 20px rgba(148, 12, 12, 0.38);

  & ${CloseButton} {
    color: white;
  }
`;

const Title = styled.h5`
  padding: 0;
  margin: 0;
  font-size: 1.2rem;
`;

export function Notifications({ children }) {
  return <NotificationList>{children}</NotificationList>;
}

export function Error({ message, onRemove }) {
  const lines = message
    .split('\n')
    .map((line, i) => <NotificationLine key={i}>{line}</NotificationLine>);

  return (
    <ErrorNotification>
      <CloseButton onClick={onRemove}>Ã—</CloseButton>
      <Title>Error</Title>
      {lines}
    </ErrorNotification>
  );
}
