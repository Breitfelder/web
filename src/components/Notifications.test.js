import React from 'react';
import { shallow } from 'enzyme';
import { Error, CloseButton } from './Notifications';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

describe('Error', () => {
  it('should render correctly', () => {
    const msg = 'Foo bar\nBaz qux';
    const wrapper = renderer.create(<Error message={msg} />);

    expect(wrapper).toMatchStyledComponentsSnapshot();
  });

  it('should call onRemove when the close button is clicked', () => {
    const spy = jest.fn();
    const wrapper = shallow(<Error message="foo" onRemove={spy} />);

    wrapper.find(CloseButton).simulate('click');
    expect(spy.mock.calls.length).toBe(1);
  });
});
