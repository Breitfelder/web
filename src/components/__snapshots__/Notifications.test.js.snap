// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error should render correctly 1`] = `
.c5 {
  padding: 0;
  margin: 0;
}

.c5 + .c5 {
  margin-top: 0.5rem;
}

.c2 {
  font-size: 1.5rem;
  position: absolute;
  right: 0.5rem;
  top: 0.2rem;
  padding: 0.3rem 0.5rem;
  margin: 0;
  border: none;
  background: transparent;
  cursor: pointer;
}

.c0 {
  position: relative;
  border-radius: 5px;
  padding: 1rem;
  margin: 0 1rem 1rem 1rem;
  background: #f44336;
  color: white;
  box-shadow: 0 4px 20px rgba(148,12,12,0.38);
}

.c0 .c1 {
  color: white;
}

.c3 {
  padding: 0;
  margin: 0;
  font-size: 1.2rem;
}

.c4 {
  max-height: calc(100vh - 80px);
  overflow: auto;
}

<div
  className="c0"
>
  <button
    className="c1 c2"
  >
    Ã—
  </button>
  <h5
    className="c3"
  >
    Error
  </h5>
  <div
    className="c4"
  >
    <p
      className="c5"
    >
      Foo bar
    </p>
    <p
      className="c5"
    >
      Baz qux
    </p>
  </div>
</div>
`;
